src/vs/workbench/contrib/positronConnections/browser/components/schemaNavigation.tsx

```diff
@@ -0,0 +1,301 @@
+					itemKey={index => entries[index].id}
+					innerRef={innerRef}
+				>
+					{ItemEntry}
+				</List>
+			</div>
+		</div>
+	);
+};
+
+interface ItemEntryProps {
+	index: number;
+	style: any;
+}
+
+interface PositronConnectionsItemProps {
+	item: IPositronConnectionEntry;
+	style?: any;
+	selected: boolean;
+
+	/**
+	 * What happens when a row is selected?
+	 */
+	onSelectedHandler: () => void;
+
+	/**
+	 * On toggle expand handler
+	 */
+	onToggleExpandHandler(id: string): void;
+}
+
+const PositronConnectionsItem = (props: React.PropsWithChildren<PositronConnectionsItemProps>) => {
+
+	// If the connection is not expandable, we add some more padding.
+	const padding = props.item.level * 10 + (props.item.expanded === undefined ? 26 : 0);
+	const handleExpand = () => {
+		props.onToggleExpandHandler(props.item.id);
+	};
+
+	const icon = (() => {
+
+		if (props.item.icon) {
+			return props.item.icon;
+		}
+
+		if (props.item.kind) {
+			// TODO: we'll probably want backends to implement the casting to a set of known
+			// types or provide their own icon.
+			switch (props.item.kind) {
+				case 'table':
+					return 'positron-table-connection';
+				case 'view':
+					return 'positron-view-connection';
+				case 'database':
+					return 'positron-database-connection';
+				case 'schema':
+					return 'positron-schema-connection';
+				case 'catalog':
+					return 'positron-catalog-connection';
+				case 'field':
+					switch (props.item.dtype) {
+						case 'character':
+							return 'positron-data-type-string';
+						case 'integer':
+						case 'numeric':
+							return 'positron-data-type-number';
+						case 'boolean':
+						case 'bool':
+							return 'positron-data-type-boolean';
+						default:
+							return 'positron-data-type-unknown';
+					}
+			}
+		}
+		// If kind is not known, then no icon is dplsayed by default.
+		return '';
+	})();
+
+	const rowMouseDownHandler = (e: MouseEvent<HTMLElement>) => {
+		// Consume the event.
+		e.preventDefault();
+		e.stopPropagation();
+
+		// Handle the event.
+		switch (e.button) {
+			// Main button.
+			case 0:
+				// TODO: handle ctrl+ click, etc.
+				props.onSelectedHandler();
+				break;
+
+			// Secondary button.
+			case 2:
+				// TODO: more options here
+				props.onSelectedHandler();
+				break;
+		}
+	};
+
+	return (
+		<div
+			className={positronClassNames(
+				'connections-item',
+				{ 'selected': props.selected }
+			)}
+			style={props.style}
+		>
+			<div className='nesting' style={{ width: `${padding}px` }}></div>
+			{
+				props.item.expanded === undefined ?
+					<></> :
+					<div
+						className='expand-collapse-area'
+						onClick={handleExpand}
+					>
+						<div
+							className={`codicon codicon-chevron-${props.item.expanded ? 'down' : 'right'}`}
+						>
+						</div>
+					</div>
+			}
+			<div
+				className='connections-details'
+				onMouseDown={rowMouseDownHandler}
+			>
+				<span className='connections-name'>{props.item.name}</span>
```

Comment:
I would write:

```
{props.item.dtype && <span className='connections-dtype'>{props.item.dtype}</span>}
{props.item.error && <span className='connections-error codicon codicon-error' title={props.item.error}></span>}
```

For theses two `span`s below:

Reply:
Resolved in 32a93e41534d7082dc61cc597bac7940027a44a4